tests/functions_tests/test_sigmoid_cross_entropy.py

```diff
@@ -0,0 +1,74 @@
+import math
+import unittest
+
+import numpy
+import six
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer.testing import attr
+
+
+if cuda.available:
+    cuda.init()
+
+
+class TestSigmoidCrossEntropy(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (4, 3)).astype(numpy.float32)
+        self.t = numpy.random.randint(0, 2, (4, 3)).astype(numpy.int32)
+
+    def check_forward(self, x_data, t_data, use_cudnn=True):
+        x_val = chainer.Variable(x_data)
+        t_val = chainer.Variable(t_data)
+        loss = functions.sigmoid_cross_entropy(x_val, t_val, use_cudnn)
+        loss_value = float(cuda.to_cpu(loss.data))
+
+        # Compute expected value
+        loss_expect = 0
+        for i in six.moves.range(self.x.shape[0]):
+            for j in six.moves.range(self.x.shape[1]):
+                xd, td = self.x[i, j], self.t[i, j]
+                xdd = 1 if self.x[i, j] >= 0 else 0
```

Comment:
You can use a bool directly as a 0/1 integer: `xdd = self.x[i, j] >= 0`. If you want to convert it into a 0/1 int explicitly, use int() instead of the if expression.